src/components/Header/Header.tsx

```diff
@@ -1,11 +1,41 @@
-import { Link } from "react-router-dom"
-import { Input, Button, Divider } from 'antd';
+import { Link, useNavigate } from "react-router-dom"
+import { Button, AutoComplete } from 'antd';
 import css from "./header.module.css"
+import { useState, useEffect, useCallback } from "react"
+import debounce from 'lodash/debounce';
+import { getGoods } from "api";
+import { Good } from "types";
+import { useAppSelector } from 'hooks/hooks';
+import { getIsAuth } from 'store/auth/selectors';
+import { getTotalCountItemsInCart } from 'store/cart/selectors';
 
-
-const { Search } = Input;
+export interface Params {
+       text: string;
+}
 
 export const Header = () => {
+       const navigate = useNavigate()
+       const isAuth = useAppSelector(getIsAuth)
+       const commonCount = useAppSelector(getTotalCountItemsInCart)
+
+       const [params, setParams] = useState<Params>({ text: "" });
+       const [goods, setGoods] = useState<Good[]>([{ categoryTypeId: "", description: "", id: "", img: "", label: "", price: "" }])
```

Comment:
начальное значение для массивов всегда пустой массив, не надо придумывать несуществующие мусорные данные
```suggestion
       const [goods, setGoods] = useState<Good[]>([])
```